src/inc/til/static_map.h

```diff
@@ -32,31 +27,30 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
         };
     }
 
-    template<typename K, typename V, typename Compare = std::less<K>, size_t N = 0, typename SortedInput = details::unsorted_input_t>
+    template<typename K, typename V, size_t N, typename SortedInput = details::unsorted_input_t>
     class static_map
     {
     public:
         using const_iterator = typename std::array<std::pair<K, V>, N>::const_iterator;
 
         template<typename... Args>
-        constexpr explicit static_map(const Args&... args) noexcept :
-            _predicate{},
-            _array{ { args... } }
+        constexpr explicit static_map(Args&&... args) noexcept :
+            _array{ { std::forward<Args>(args)... } }
         {
             static_assert(sizeof...(Args) == N);
             if constexpr (!SortedInput::value)
             {
-                const auto compareKeys = [&](const auto& p1, const auto& p2) { return _predicate(p1.first, p2.first); };
-                std::sort(_array.begin(), _array.end(), compareKeys); // compile-time sorting starting C++20
+                std::sort(_array.begin(), _array.end());
             }
         }
 
-        [[nodiscard]] constexpr const_iterator find(const K& key) const noexcept
+        [[nodiscard]] constexpr const_iterator find(const auto& key) const noexcept
         {
-            const auto compareKey = [&](const auto& p) { return _predicate(p.first, key); };
-            const auto iter{ std::partition_point(_array.begin(), _array.end(), compareKey) };
+            const auto iter = std::partition_point(_array.begin(), _array.end(), [&](const auto& p) {
+                return p.first < key;
+            });

```

Comment:
Passing the closure by value produces better assembly, because `partition_point` takes it by value.